### DISCUSSION TITLE
Different ways of pushing commits from template repo

### DISCUSSION THREAD

**Discussion by george-gca** [AndreasAugustin/actions-template-sync#319] [Ideas]
Parting from #316, I believe it would be useful to make squashing commits an option, and maybe also add new options, like `rebase`.

Currently my repositories look like this:
- [main template repo](https://github.com/alshedivat/al-folio) - awesome website template, but missing a couple of things to suit my needs
- [my template repo](https://github.com/george-gca/multi-language-al-folio) - a fork of the main template repo, with some added features. Every now and then I rebase it with the main template repo, to keep everything in sync
- my website - created from my template repo (not as a fork). Ideally, every now and then I would like to rebase it with my template repo

Since all the changes that I introduced have to be applied in the latest version of the previous repo, it makes sense for me to keep rebasing it (also to keep the history as linear as possible). For this reason, I believe a `rebase` option would be nice.

**Comment by AndreasAugustin** [comment]
The responsible line is within [src/sync_template.sh](https://github.com/AndreasAugustin/actions-template-sync/blob/main/src/sync_template.sh) line 72.

![image](https://user-images.githubusercontent.com/8027933/219946844-73383521-01b2-4b0f-9cce-1be8e19387c8.png)

an idea would be to make thw whole pull arguments configurable.
Default would be

```bash
--allow-unrelated-histories --squash --strategy=recursive -X theirs
```

That way the user would have whole freedom to choose whatever strategy she/he need to use.

Remark:
With a rebase the history would be nicer but solving the merge conflicts is more hustle. At least some advanced git knowledge is needed. But the user would be able to choose it on her/his own

**Comment by AndreasAugustin** [comment]
@george-gca would be a parameter to configure the pull behaviour sufficient? Then you have the full freedom to set for your use case.

**Reply by george-gca** [reply]
That's exactly what I thought of. Something like this:

```bash
# PARAMS with default values
PARAMS="--allow-unrelated-histories --squash --strategy=recursive"
git pull "${SOURCE_REPO}" $PARAMS -X theirs
```

**Reply by AndreasAugustin** [reply]
will go with that solution

**Comment by AndreasAugustin** [comment]
Related PR: #330 
and further info in #316
